src/argilla/client/client.py

```diff
@@ -228,13 +246,15 @@ def get_workspace(self) -> str:
         return self.http_client.headers.get(WORKSPACE_HEADER_NAME)
 
     def list_workspaces(self) -> List[WorkspaceModel]:
-        """Lists all the availble workspaces for the current user.
+        """Lists all the available workspaces for the current user.
 
         Returns:
             A list of `WorkspaceModel` objects, containing the workspace
             attributes: name, id, created_at, and updated_at.
         """
-        return workspaces_api.list_workspaces(client=self.http_client.httpx).parsed
+        user_workspaces = users_api.whoami(self.http_client).workspaces
```

Comment:
TODO: at some point we should also refactor the `whoami` to use the `httpx.Client` instead and return the response, instead of directly the parsed response, plus we should add proper error handling